code/modules/atmospherics/environmental/LINDA_turf_tile.dm

```diff
@@ -84,9 +84,11 @@
 /turf/open/return_analyzable_air()
 	return return_air()
 
-/turf/temperature_expose()
-	if(temperature > heat_capacity)
-		to_be_destroyed = TRUE
+/turf/temperature_expose(datum/gas_mixture/air, exposed_temperature, exposed_volume)
+	if(exposed_temperature > max_fire_temperature_sustained)
+		max_fire_temperature_sustained = exposed_temperature
+	if(max_fire_temperature_sustained > heat_capacity)
+		set_heat_damage_amount(exposed_temperature)
```

Comment:
Every turf on fire will call two range calls every atmos tick seems pretty extreme. Was this profiled.

This is also overridden for plasma turf types iirc.

Reply:
i did some testing on local and i didnt see any performance issues. For big plasma fires the atmos ticks get slowed down a lot by the gases themselfs and this still doesnt affect the performances.

For the plasma turfs yes they override it, any way that i can use this on them and their thing too? adding .=..() before what they do is enough?

Reply:
that's not really a good way to profile, you should use the profiler.

Reply:
no i mean, i used the profiler to check the performances, there where no issues with performances and everything seems to behave properly